spacy/util.py

```diff
@@ -1032,12 +1032,12 @@ def make_tempdir() -> Generator[Path, None, None]:
 
     YIELDS (Path): The path of the temp directory.
     """
-    d = Path(tempfile.mkdtemp())
-    yield d
+
     try:
-        shutil.rmtree(str(d))
+        with tempfile.TemporaryDirectory() as td:
+            yield Path(td)
     except PermissionError as e:
-        warnings.warn(Warnings.W091.format(dir=d, msg=e))
+        warnings.warn(Warnings.W091.format(dir=td, msg=e))
```

Comment:
Can you make a separate PR with this change + explanation?

Reply:
Here's the PR 

https://github.com/explosion/spaCy/pull/12285